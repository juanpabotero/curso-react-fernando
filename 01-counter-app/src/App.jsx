function App() {
  return <h1>Hola desde el AppComponent</h1>;
}

export default App;
