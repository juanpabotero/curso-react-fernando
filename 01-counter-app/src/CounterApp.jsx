import { useState } from "react";
import PropTypes from "prop-types";

export const CounterApp = ({ value = 0 }) => {
  const [counter, setCounter] = useState(value);

  const handleAdd = () => {
    setCounter(counter + 1);
    // tambien podria hacerlo asi:
    // setCounter((c) => c + 1);
  };
  const handleSubstract = () => setCounter(counter - 1);
  const handleReset = () => setCounter(value);

  return (
    <section className="mt-8">
      <h1 className="text-2xl font-bold">CounterApp</h1>
      <h2 className="text-xl">{counter}</h2>
      <div className="flex justify-center gap-1 mt-2">
        <button
          onClick={handleAdd}
          className="py-1 px-2 bg-slate-600 rounded-md hover:bg-slate-400 transition-all"
        >
          +1
        </button>
        <button
          onClick={handleSubstract}
          className="py-1 px-2 bg-slate-600 rounded-md hover:bg-slate-400 transition-all"
        >
          -1
        </button>
        <button
          aria-label="btn-reset"
          onClick={handleReset}
          className="py-1 px-2 bg-slate-600 rounded-md hover:bg-slate-400 transition-all"
        >
          Reset
        </button>
      </div>
    </section>
  );
};

CounterApp.propTypes = {
  value: PropTypes.number,
};
