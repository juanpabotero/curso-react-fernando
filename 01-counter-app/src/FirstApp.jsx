import PropTypes from "prop-types";

/* Lo que no va a cambiar cuando se use el componente,
lo puedo poner afuera de la función principal para evitar
que se vuelva a ejecutar esa parte que no va a cambiar */
const getTitle = (title) => {
  return title;
};

/* props son las propiedades que recibe el componente,  
  es un objeto, pero normalmente se desestructura para usar dentro  
  de la función. Sirven para pasar información del componente  
  padre al componente hijo. */
export const FirstApp = ({ name, age }) => {
  return (
    /* esto es un fragmento y se usa para devolver varios elementos */
    <>
      {/* Puedo usar {} para poner expresiones validas de JavaScript,  
      recordar que las expresiones devuelven un valor */}
      <h1>{getTitle("Este es el FirstAppComponent")}</h1>
      <p>
        Hola, mi nombre es {name} y tengo {age}
      </p>
    </>
  );
};

/* definir los tipos para las props */
FirstApp.propTypes = {
  age: PropTypes.number.isRequired,
  name: PropTypes.string.isRequired,
};

/* definir los valores por defecto */
FirstApp.defaultProps = {
  age: 0,
  name: "No hay nombre",
};
