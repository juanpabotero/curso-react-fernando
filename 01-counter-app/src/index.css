@tailwind base;
@tailwind components;
@tailwind utilities;

html,
body {
  background-color: black;
  min-height: 100%;
}
