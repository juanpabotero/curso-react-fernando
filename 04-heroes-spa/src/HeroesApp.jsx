import { AuthProvider } from './auth';
import { AppRouter } from './router/AppRouter';

export const HeroesApp = () => {
  return (
    // provee la informaci√≥n a todos los componentes hijos
    <AuthProvider>
      {/* mostrar las paginas segun las rutas definidas */}
      <AppRouter />
    </AuthProvider>
  );
};
