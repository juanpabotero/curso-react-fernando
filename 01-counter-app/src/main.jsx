/* ESTE ES EL PUNTO DE ENTRADA DE LA APLICACIÃ“N */

import React from "react";
import ReactDOM from "react-dom/client";
import App from "./App";
import { FirstApp } from "./FirstApp";
import { CounterApp } from "./CounterApp";
import "./index.css";

ReactDOM.createRoot(document.getElementById("root")).render(
  <React.StrictMode>
    <main className="container mx-auto min-h-screen text-white py-10 px-4 text-center">
      <App />
      <FirstApp name="Juan Pablo" age={29} />
      <CounterApp value={10} />
    </main>
  </React.StrictMode>
);
