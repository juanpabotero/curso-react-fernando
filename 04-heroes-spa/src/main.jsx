import React from 'react';
import ReactDOM from 'react-dom/client';
import { BrowserRouter } from 'react-router-dom';

import { HeroesApp } from './HeroesApp';
import './styles.css';

ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    {/* para el funcionamiento de las rutas en la web */}
    <BrowserRouter>
      <HeroesApp />
    </BrowserRouter>
  </React.StrictMode>
);
