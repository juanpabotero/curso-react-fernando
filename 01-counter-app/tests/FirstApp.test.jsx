import { describe, expect, test } from "vitest";
import { render } from "@testing-library/react";
import { FirstApp } from "../src/FirstApp";

describe("Pruebas en <FirstApp />", () => {
  /* este test le toma una foto al componente y lo compara con lo que 
  se renderice en este test */
  test("Debe hacer match con el snapshot", () => {
    /* renderiza el componente */
    const { container } = render(<FirstApp name="Juan Pablo" age={29} />);
    expect(container).toMatchSnapshot();
  });

  test("Debe mostrar el titulo en h1", () => {
    const title = "Este es el FirstAppComponent";
    /* renderiza el componente */
    const { container, getByText, getByTestId } = render(
      <FirstApp name="Juan Pablo" age={29} />
    );
    expect(getByText(title)).toBeTruthy();
    const h1 = container.querySelector("h1");
    expect(h1.innerHTML).toContain("Este es el FirstAppComponent");
    expect(getByTestId("test-title").innerHTML).toContain(title);
  });

  test("Debe mostrar el nombre enviado por props", () => {
    const name = "Juan Pablo";
    const age = 29;
    const { findByText } = render(
      <FirstApp name={name} age={age} />
    );
    expect(findByText(name)).toBeTruthy();
  });
});
