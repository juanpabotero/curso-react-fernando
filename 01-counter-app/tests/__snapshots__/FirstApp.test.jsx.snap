// Vitest Snapshot v1

exports[`Pruebas en <FirstApp /> > Debe hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Este es el FirstAppComponent
  </h1>
  <p>
    Hola, mi nombre es 
    Juan Pablo
     y tengo 
    29
  </p>
</div>
`;
